Content.Server/GameObjects/Components/Fluids/PuddleComponent.cs

```diff
@@ -315,42 +308,59 @@ private void CheckOverflow()
             SpillToNeighbours(neighborPuddles, spillAmount);
         }
 
-        private void SpillToNeighbours(IEnumerable<IEntity> neighbors, int spillAmount)
+        // TODO: Move the below to SnapGrid?
+        /// <summary>
+        /// Will yield a random direction until none are left
+        /// </summary>
+        /// <returns></returns>
+        private static IEnumerable<Direction> RandomDirections()
         {
-            foreach (var neighborPuddle in neighbors)
+            var directions = new List<Direction>
```

Comment:
If you want you can optimize this slightly by initializing an array instead of a list. End result should be the same but you'll have less allocations from filling the list.

Reply:
Cheers mate, will keep in mind for the future.